internal/gengapic/markdown_test.go

```diff
@@ -73,11 +73,20 @@ func TestMDPlain(t *testing.T) {
 			in:   "html <a href=\"/link/to/some#thing\">\n with a softbreak</a> <br> test",
 			want: "html with a softbreak (at /link/to/some#thing) \n test",
 		},
+		{
+			in:   "html <a \n  href=\"/link/to/some#thing\">\n with a linebreak in tag</a> <br> test",
+			want: "html with a linebreak in tag (at /link/to/some#thing) \n test",
+		},
 		{
 			in:   "link to [a search engine](https://www.google.com) with request type [Search][foo.bar_baz.v1.Search], [biz][][buz][baz].",
 			want: "link to a search engine (at https://www.google.com) with request type Search, bizbuz.",
 		},
 	} {
+		defer func() {
+			if r := recover(); r != nil {
```

Comment:
Why is this necessary?

Reply:
Failure to match the regex results in a panic, adding this `Errorf` output allows you to see which test caused it.

Reply:
It's a bit of a tradeoff since with this statement you don't get the whole backtrace, just the message for the panic.

#### Before

```
--- FAIL: TestMDPlain (0.00s)
panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]

goroutine 130 [running]:
testing.tRunner.func1.2({0x163c560, 0xc000042c00})
	/usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x163c560, 0xc000042c00})
	/usr/local/go/src/runtime/panic.go:838 +0x207
github.com/googleapis/gapic-generator-go/internal/gengapic.(*mdRenderer).html(0xc0004cc000, 0xc0006a16b0)
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:129 +0x257
github.com/googleapis/gapic-generator-go/internal/gengapic.(*mdRenderer).plain(0xc0004cc000, {0x1777820?, 0xc0006a16b0?})
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:91 +0x2d3
github.com/googleapis/gapic-generator-go/internal/gengapic.(*mdRenderer).plain(0xc0004cc000, {0x17779d0?, 0xc0004cfd40?})
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:59 +0x8fe
github.com/googleapis/gapic-generator-go/internal/gengapic.mdPlain({0x16a2123, 0x49})
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown.go:35 +0x11f
github.com/googleapis/gapic-generator-go/internal/gengapic.TestMDPlain(0xc000702d00)
	/Users/chrisdsmith/oss/gapic-generator-go/internal/gengapic/markdown_test.go:85 +0xa5
testing.tRunner(0xc000702d00, 0x16c3848)
	/usr/local/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1486 +0x35f
FAIL	github.com/googleapis/gapic-generator-go/internal/gengapic	0.788s
FAIL
make: *** [test-gapic] Error 1
```


#### After

```
--- FAIL: TestMDPlain (0.00s)
    markdown_test.go:87: html <ahref="/link/to/some#thing">
         with a linebreak in tag</a> <br> test resulted in a panic: runtime error: index out of range [-1]
FAIL
FAIL	github.com/googleapis/gapic-generator-go/internal/gengapic	0.405s
FAIL
make: *** [test-gapic] Error 1
```

Reply:
I see, I see...would it be too much to plumb an `error` through the call stack? WDYT?

Reply:
Let me see about that! 👍

Reply:
You could also consider that a separate improvement and do it in another PR

Reply:
"almost done"

Reply:
It turns out to be A LOT of plumbing, since the stack is pretty deep and not currently returning errors. Do you think it's a good idea to add this, or is it better to just debug the panic in case of a malformed `<a>` element?

Reply:
I figured it might be gnarly...let's just debug the panic. Technically, HTML is banned in our API artifact documentation, so let's not spend more time on it. Thanks for looking anyways!

Reply:
Updated to remove the `recover`, thanks!